python/taichi/misc/image.py

```diff
@@ -16,15 +16,37 @@ def imwrite(img, filename):
     assert len(img.shape) in [2,
                               3], "Image must be either RGB/RGBA or greyscale"
 
-    resx, resy = img.shape[:2]
     if len(img.shape) == 2:
-        comp = 1
+        img = img.reshape(*img.shape, 1)
+
+    assert img.shape[2] in [1, 3, 4], "Image must be either RGB/RGBA or greyscale"
+
+    return img.swapaxes(0, 1)[::-1, :]
+
+
+
+def imdisplay(img):
+    """
+    Try to display image in interactive shell.
+    """
+    if ti.lang.shell.oinspect.name.startswith('IPython'):
+        import PIL.Image
+        from io import BytesIO
+        import IPython.display
+        import numpy as np
+        img = imcook(img)
+        f = BytesIO()
+        PIL.Image.fromarray(img).save(f, 'png')
+        IPython.display.display(IPython.display.Image(data=f.getvalue()))
```

Comment:
I'm not familiar with `IPython.display`, but it might be safer to wrap the bytes with a context manager so it gets closed automatically?

e.g.
```suggestion
        with BytesIO() as f:
            PIL.Image.fromarray(img).save(f, 'png')
            IPython.display.display(IPython.display.Image(data=f.getvalue()))
```